Pacts from broker are downloaded as HTML

Environment: 
- Broker with docker image dius/pact_broker:1.1.0, and also dius/pact-broker:2.0.0-beta
- JDK 1.8.0_111
- Java(TM) SE Runtime Environment (build 1.8.0_111-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.111-b14, mixed mode)

- Plugin configuration:
```
        <plugin>
            <groupId>com.github.warmuuh</groupId>
            <artifactId>pactbroker-maven-plugin</artifactId>
            <version>0.0.10</version>
            <configuration>
                <provider>foo</provider>
                <brokerUrl>http://localhost</brokerUrl>
            </configuration>
        </plugin>
```

In the broker, I have a pact with the provider 'foo', tagged with version 0.0.0.
When executing plugin, I have the following log:

```
[INFO] >>> pactbroker-maven-plugin:0.0.10:download-pacts (default-cli) > generate-test-resources @ provider >>>
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ provider ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] Copying 0 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ provider ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] <<< pactbroker-maven-plugin:0.0.10:download-pacts (default-cli) < generate-test-resources @ provider <<<
[INFO] 
[INFO] --- pactbroker-maven-plugin:0.0.10:download-pacts (default-cli) @ provider ---
[INFO] Downloading pact links from http://localhost/pacts/provider/foo/latest
[INFO] Downloading pact from http://localhost/pacts/provider/foo/consumer/foo_consumer/version/0.0.0
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.435 s
[INFO] Finished at: 2017-05-02T13:20:19+02:00
[INFO] Final Memory: 28M/514M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.warmuuh:pactbroker-maven-plugin:0.0.10:download-pacts (default-cli) on project provider: Failed to download pacts: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 2 column 14 -> [Help 1]
```

The cause is that the requests sent to the broker always have the header 'Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2' (I've seen that with a sniffer), so if possible, the broker returns the HTML view instead of the needed JSON.

In the code of BrokerRepositoryProvider.java, the connection has the right 'Content-Type' but 'Accept' is not defined, I don't know where that value comes.

--- a/src/main/java/com/github/wrm/pact/repository/BrokerRepositoryProvider.java
+++ b/src/main/java/com/github/wrm/pact/repository/BrokerRepositoryProvider.java
@@ -1,5 +1,11 @@
 package com.github.wrm.pact.repository;
 
+import com.github.wrm.pact.domain.PactFile;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import org.apache.maven.plugin.logging.Log;
+
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOException;
@@ -14,14 +20,6 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 import java.util.Scanner;
-
-import org.apache.commons.codec.binary.Base64;
-import org.apache.maven.plugin.logging.Log;
-
-import com.github.wrm.pact.domain.PactFile;
-import com.google.gson.JsonArray;
-import com.google.gson.JsonElement;
-import com.google.gson.JsonParser;
 
 import static org.apache.commons.codec.binary.Base64.encodeBase64URLSafeString;
 
@@ -76,7 +74,7 @@ public class BrokerRepositoryProvider implements RepositoryProvider {
 
         URL url = new URL(path);
         HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-        connection.setRequestProperty("Content-Type", "application/json");
+        connection.setRequestProperty("Accept", "application/json");
         connection.setDoInput(true);
         connection.setDoOutput(true);
         addBasicAuthTo(connection);
@@ -113,6 +111,7 @@ public class BrokerRepositoryProvider implements RepositoryProvider {
         connection.setDoOutput(true);
         connection.setDoInput(true);
         connection.setRequestMethod("PUT");
+        connection.setRequestProperty("Accept", "application/json");
         connection.setRequestProperty("Content-Type", "application/json");
         connection.setRequestProperty("charset", StandardCharsets.UTF_8.displayName());
         addBasicAuthTo(connection);
@@ -156,7 +155,7 @@ public class BrokerRepositoryProvider implements RepositoryProvider {
             FileNotFoundException {
         URL url = new URL(link);
         HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-        connection.setRequestProperty("Content-Type", "application/json");
+        connection.setRequestProperty("Accept", "application/json");
         connection.setDoInput(true);
         addBasicAuthTo(connection);
 

--- a/src/test/java/com/github/wrm/pact/repository/BrokerRepositoryProviderTest.java
+++ b/src/test/java/com/github/wrm/pact/repository/BrokerRepositoryProviderTest.java
@@ -1,18 +1,12 @@
 package com.github.wrm.pact.repository;
 
-import static java.util.Optional.empty;
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.MatcherAssert.assertThat;
-
-import java.io.File;
-import java.io.PrintWriter;
-import java.nio.charset.StandardCharsets;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-
+import au.com.dius.pact.consumer.ConsumerPactBuilder.PactDslWithProvider.PactDslWithState;
+import au.com.dius.pact.consumer.Pact;
+import au.com.dius.pact.consumer.PactRule;
+import au.com.dius.pact.consumer.PactVerification;
+import au.com.dius.pact.model.PactFragment;
+import com.github.wrm.pact.OpenPortProvider;
+import com.github.wrm.pact.domain.PactFile;
 import org.apache.maven.plugin.logging.SystemStreamLog;
 import org.junit.Before;
 import org.junit.Rule;
@@ -20,14 +14,14 @@ import org.junit.Test;
 import org.junit.rules.ExpectedException;
 import org.junit.rules.TemporaryFolder;
 
-import au.com.dius.pact.consumer.ConsumerPactBuilder.PactDslWithProvider.PactDslWithState;
-import au.com.dius.pact.consumer.Pact;
-import au.com.dius.pact.consumer.PactRule;
-import au.com.dius.pact.consumer.PactVerification;
-import au.com.dius.pact.model.PactFragment;
+import java.io.File;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
 
-import com.github.wrm.pact.OpenPortProvider;
-import com.github.wrm.pact.domain.PactFile;
+import static java.util.Optional.empty;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
 
 public class BrokerRepositoryProviderTest {
 
@@ -131,8 +125,8 @@ public class BrokerRepositoryProviderTest {
         return builder
                 .uponReceiving("a pact file")
                 .path("/pacts/provider/" + PROVIDER_NAME + "/consumer/" + CONSUMER_NAME + "/version/"
-                        + CONSUMER_VERSION).body(pactJson).headers(getHeaders()).method("PUT").willRespondWith()
-                .headers(getHeaders()).status(201).body(pactJson)
+                        + CONSUMER_VERSION).body(pactJson).headers(getWriteRequestHeaders()).method("PUT").willRespondWith()
+                .headers(getResponseHeaders()).status(201).body(pactJson)
                 .toFragment();
     }
 
@@ -143,52 +137,65 @@ public class BrokerRepositoryProviderTest {
         return builder
                 .uponReceiving("a pact file")
                 .path("/pacts/provider/" + PROVIDER_NAME + "/consumer/" + CONSUMER_NAME + "/version/"
-                        + CONSUMER_VERSION).body(pactJson).headers(getHeaders()).method("PUT").willRespondWith()
-                .headers(getHeaders()).status(201).body(pactJson)
+                        + CONSUMER_VERSION).body(pactJson).headers(getWriteRequestHeaders()).method("PUT").willRespondWith()
+                .headers(getResponseHeaders()).status(201).body(pactJson)
                 .uponReceiving("a pact tagging request")
-                .path("/pacticipants/" + CONSUMER_NAME + "/versions/" + CONSUMER_VERSION + "/tags/" + TAG_NAME.get()).headers(getHeaders()).method("PUT").willRespondWith()
-                .headers(getHeaders()).status(201)
+                .path("/pacticipants/" + CONSUMER_NAME + "/versions/" + CONSUMER_VERSION + "/tags/" + TAG_NAME.get()).headers(getResponseHeaders()).method("PUT").willRespondWith()
+                .headers(getResponseHeaders()).status(201)
                 .toFragment();
     }
 
     @Pact(state = "pact-already-uploaded", provider = "broker-maven-plugin", consumer = "pact-broker")
     public PactFragment createFragmentForUploadingPact(PactDslWithState builder) {
 
-        return builder.uponReceiving("an already existing pact file").path(pactPath).headers(getHeaders())
-                .method("PUT").willRespondWith().headers(getHeaders()).status(200).body(pactJson).toFragment();
+        return builder.uponReceiving("an already existing pact file").path(pactPath).headers(getWriteRequestHeaders())
+                .method("PUT").willRespondWith().headers(getResponseHeaders()).status(200).body(pactJson).toFragment();
     }
 
     @Pact(state = "one-provider-pact-link-present", provider = "broker-maven-plugin", consumer = "pact-broker")
     public PactFragment createFragmentForDownloadingPactLinks(PactDslWithState builder) {
 
         return builder.uponReceiving("a request for the latest provider pacts")
-                .path("/pacts/provider/" + PROVIDER_NAME + "/latest").headers(getHeaders()).method("GET")
-                .willRespondWith().headers(getHeaders()).status(200).body(providerJson).toFragment();
+                .path("/pacts/provider/" + PROVIDER_NAME + "/latest").headers(getReadRequestHeaders()).method("GET")
+                .willRespondWith().headers(getResponseHeaders()).status(200).body(providerJson).toFragment();
     }
 
     @Pact(state = "one-prod-provider-pact-link-present", provider = "broker-maven-plugin", consumer = "pact-broker")
     public PactFragment createFragmentForDownloadingPactLinksForProdTag(PactDslWithState builder) {
 
         return builder.uponReceiving("a request for the latest provider pacts for the prod tag")
-                .path("/pacts/provider/" + PROVIDER_NAME + "/latest/prod").headers(getHeaders()).method("GET")
-                .willRespondWith().headers(getHeaders()).status(200).body(providerJson).toFragment();
+                .path("/pacts/provider/" + PROVIDER_NAME + "/latest/prod").headers(getReadRequestHeaders()).method("GET")
+                .willRespondWith().headers(getResponseHeaders()).status(200).body(providerJson).toFragment();
     }
 
     @Pact(state = "one-pact-present", provider = "broker-maven-plugin", consumer = "pact-broker")
     public PactFragment createFragmentForDownloadingPact(PactDslWithState builder) {
 
-        return builder.uponReceiving("a request for the latest provider pacts").path(pactPath).headers(getHeaders())
-                .method("GET").willRespondWith().headers(getHeaders()).status(200).body(pactJson).toFragment();
+        return builder.uponReceiving("a request for the latest provider pacts").path(pactPath).headers(getReadRequestHeaders())
+                .method("GET").willRespondWith().headers(getResponseHeaders()).status(200).body(pactJson).toFragment();
     }
 
     @Pact(state = "provider-no-pact-link-present", provider = "broker-maven-plugin", consumer = "pact-broker")
     public PactFragment createFragmentForDownloadingPactLinksOfProviderWithNoPactLinkPresent(PactDslWithState builder) {
         return builder.uponReceiving("a request for the latest pacts of a provider with no link present")
-                .path("/pacts/provider/provider-no-pact-link-present/latest").headers(getHeaders()).method("GET")
-                .willRespondWith().headers(getHeaders()).status(404).toFragment();
+                .path("/pacts/provider/provider-no-pact-link-present/latest").headers(getReadRequestHeaders()).method("GET")
+                .willRespondWith().headers(getResponseHeaders()).status(404).toFragment();
     }
 
-    private Map<String, String> getHeaders() {
+    private Map<String, String> getReadRequestHeaders() {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Accept", "application/json");
+        return headers;
+    }
+
+    private Map<String, String> getWriteRequestHeaders() {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Accept", "application/json");
+        headers.put("Content-Type", "application/json");
+        return headers;
+    }
+
+    private Map<String, String> getResponseHeaders() {
         Map<String, String> headers = new HashMap<>();
         headers.put("Content-Type", "application/json");
         return headers;

